### DISCUSSION TITLE
onDelete - a callback for image deletion - 🙏🏼

### DISCUSSION THREAD

**Discussion by ScottAgirs** [outline/rich-markdown-editor#471] [Ideas]
Hey guys, I saw the issue #167, but since that one was closed...

IMHO, a callback whenever an image is deleted would be super useful.

Here is why I think you should consider this
- it is obviously a common pattern
- [as per this comment](https://github.com/outline/rich-markdown-editor/issues/167#issuecomment-621542206) comparing could be an option, but even such simple of a scenario, when user uploads and deletes right away (without saving) quickly become a nightmarish to keep track of and comparing...

Instead if we'd have the hook, it would provide a convenient and common API pattern.

I live this library and have completely switched from the idea to further develop my own based on Slate.js, but this has been a big issue, fingers crossed 🤞🏼

**Comment by tommoor** [comment]
If it was PR'd I think it could be accepted alongside a warning not to rely on it to detect deleted images due to the aforementioned limitations

**Reply by ScottAgirs** [reply]
Hey @tommoor , thanks for your reply, just to clarify - are you saying you're open to the idea given that there's a warning before deleting? I think a warning or some sort of layer of auxiliary state that holds the image source in state and if the user does the undo, uploads it back again, perhaps?

**Reply by tommoor** [reply]
I mean a warning for any developers using the callback. Just stating that it isn't a foolproof way to ensure that images deleted in the editor are deleted on the backend :)

**Reply by ScottAgirs** [reply]
Yeah, honestly, I had exact same idea in mind. Did someone ever pursued the PR? If I can help w it - happy to jump on!

**Reply by tommoor** [reply]
Nope, it hasn't come up as a request again until now
